src/systemd.c

```diff
@@ -100,13 +100,15 @@ int systemd_init(void) {
 		close(systemd_fd);
 		systemd_fd = -1;
 	}
+	free((char *)sd_unix_path);
 	sd_unix_path = NULL;
 	return -1;
 }
 
 void systemd_free(void) {
 	if (systemd_fd > 0)
 		close(systemd_fd);
+	free((char *)sd_unix_path);
```

Comment:
Nit: Let's define `sd_unix_path` as non-const instead of casting it here?

Reply:
These were my initial thoughts as well, but I think it's better to leave `sd_unix_path` a `const char *`. In this particular case, it's a global variable, unlike other places where I turned `const char *` to plain `char *`.